sim/hunter/talents.go

```diff
@@ -160,3 +162,11 @@ func (hunter *Hunter) mortalShots() float64 {
 func (hunter *Hunter) trapMastery() float64 {
 	return 5 * float64(hunter.Talents.TrapMastery) * core.SpellHitRatingPerHitChance
 }
+
+func (hunter *Hunter) applyEfficiency() {
+	hunter.OnSpellRegistered(func(spell *core.Spell) {
+		if spell.Flags.Matches(SpellFlagShot) || spell.Flags.Matches(SpellFlagStrike) && spell.SpellCode != SpellCode_HunterKillShot {
```

Comment:
I don't think this is applying properly to stings and I also noticed that wowhead lists Volley as an affected spell as well https://www.wowhead.com/classic/spell=19420/efficiency

Reply:
Also I think you may need to do parentheses around the `||` condition, otherwise order of operations will evaluate the statement as `X || (Y && Z)` instead of `(X || Y) && Z`

Reply:
decided to just move it out of that if statement, basically made an if-statement for exceptions like kill shot and an if-statement for the correct spell flags + volley